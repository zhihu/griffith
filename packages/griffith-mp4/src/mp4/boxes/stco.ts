import Stream from '../stream'

export default function stco(buffer) {
  const stream = new Stream(buffer)

  const version = stream.readByte(1)
  const flags = stream.readByte(3)

  const entryCount = stream.readByte(4)
  const samples = []

  for (let i = 0; i < entryCount; i++) {
    samples.push({chunkOffset: stream.readByte(4)})
  }

  const stcoBox = {
    version,
    flags,
    samples,
  }

  return stcoBox
}
