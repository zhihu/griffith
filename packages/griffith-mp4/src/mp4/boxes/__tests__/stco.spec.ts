import stco from '../stco'

describe('stco box', () => {
  it('should get the stco box tree', () => {
    // prettier-ignore
    const stcoBody = new Uint8Array([
      0x00, 0x00, 0x00, 0x00,
      0x00, 0x00, 0x00, 0xFB,
      0x00, 0x00, 0x24, 0x37,
      0x00, 0x00, 0x42, 0xCC,
      0x00, 0x00, 0x49, 0xDB,
      0x00, 0x00, 0x4C, 0xE7,
      0x00, 0x00, 0x50, 0x8C,
      0x00, 0x00, 0x53, 0xF9,
      0x00, 0x00, 0x56, 0x25,
      0x00, 0x00, 0x61, 0x36,
      0x00, 0x00, 0x67, 0x3F,
      0x00, 0x00, 0x70, 0x94,
      0x00, 0x00, 0x75, 0x53,
      0x00, 0x00, 0x78, 0x7C,
      0x00, 0x00, 0x81, 0x77,
      0x00, 0x00, 0x86, 0x2E,
      0x00, 0x00, 0x90, 0x38,
      0x00, 0x00, 0x98, 0x96,
      0x00, 0x00, 0xA1, 0x46,
      0x00, 0x00, 0xA4, 0x35,
      0x00, 0x00, 0xAE, 0x3B,
      0x00, 0x00, 0xB3, 0x64,
      0x00, 0x00, 0xBC, 0xF8,
      0x00, 0x00, 0xBF, 0xF7,
      0x00, 0x00, 0xC7, 0xCC,
      0x00, 0x00, 0xD0, 0x7E,
      0x00, 0x00, 0xD3, 0x3C,
      0x00, 0x00, 0xE1, 0x45,
      0x00, 0x00, 0xE5, 0xDB,
      0x00, 0x00, 0xF4, 0xC8,
      0x00, 0x00, 0xF9, 0x17,
      0x00, 0x01, 0x07, 0x8E,
      0x00, 0x01, 0x0E, 0x08,
      0x00, 0x01, 0x1A, 0xBF,
      0x00, 0x01, 0x1F, 0x53,
      0x00, 0x01, 0x2D, 0x7A,
      0x00, 0x01, 0x33, 0xAF,
      0x00, 0x01, 0x3E, 0x38,
      0x00, 0x01, 0x41, 0x86,
      0x00, 0x01, 0x51, 0x26,
      0x00, 0x01, 0x56, 0xBC,
      0x00, 0x01, 0x65, 0x07,
      0x00, 0x01, 0x72, 0xFC,
      0x00, 0x01, 0x79, 0xAA,
      0x00, 0x01, 0x7C, 0x7F,
      0x00, 0x01, 0x81, 0x9E,
      0x00, 0x01, 0x8D, 0xF8,
      0x00, 0x01, 0x93, 0x07,
      0x00, 0x01, 0x96, 0x67,
      0x00, 0x01, 0xA5, 0x9E,
      0x00, 0x01, 0xB0, 0xAE,
      0x00, 0x01, 0xBA, 0x56,
      0x00, 0x01, 0xC5, 0x6F,
      0x00, 0x01, 0xD1, 0x39,
      0x00, 0x01, 0xE3, 0x5D,
      0x00, 0x01, 0xE8, 0x6B,
      0x00, 0x01, 0xFB, 0x52,
      0x00, 0x02, 0x01, 0xF2,
      0x00, 0x02, 0x12, 0xA6,
      0x00, 0x02, 0x17, 0x21,
      0x00, 0x02, 0x1B, 0xA4,
      0x00, 0x02, 0x2C, 0x65,
      0x00, 0x02, 0x33, 0x39,
      0x00, 0x02, 0x38, 0x74,
      0x00, 0x02, 0x3B, 0x4B,
      0x00, 0x02, 0x4E, 0x70,
      0x00, 0x02, 0x53, 0x74,
      0x00, 0x02, 0x5F, 0x19,
      0x00, 0x02, 0x63, 0xC6,
      0x00, 0x02, 0x6D, 0xD8,
      0x00, 0x02, 0x7F, 0x3E,
      0x00, 0x02, 0x8D, 0x61,
      0x00, 0x02, 0x9F, 0x34,
      0x00, 0x02, 0xA4, 0x9A,
      0x00, 0x02, 0xB5, 0x8A,
      0x00, 0x02, 0xC2, 0xB8,
      0x00, 0x02, 0xC9, 0x64,
      0x00, 0x02, 0xCD, 0x59,
      0x00, 0x02, 0xD8, 0x8B,
      0x00, 0x02, 0xEC, 0xF0,
      0x00, 0x02, 0xF2, 0xAE,
      0x00, 0x03, 0x06, 0x1F,
      0x00, 0x03, 0x0E, 0xDF,
      0x00, 0x03, 0x1F, 0xA0,
      0x00, 0x03, 0x25, 0x17,
      0x00, 0x03, 0x3E, 0x01,
      0x00, 0x03, 0x48, 0x62,
      0x00, 0x03, 0x59, 0x11,
      0x00, 0x03, 0x5C, 0x88,
      0x00, 0x03, 0x6F, 0x73,
      0x00, 0x03, 0x76, 0x7E,
      0x00, 0x03, 0x89, 0x6D,
      0x00, 0x03, 0x91, 0x14,
      0x00, 0x03, 0x9F, 0x31,
      0x00, 0x03, 0xA5, 0x32,
      0x00, 0x03, 0xB2, 0xF7,
      0x00, 0x03, 0xBA, 0x47,
      0x00, 0x03, 0xC6, 0xF5,
      0x00, 0x03, 0xCC, 0x18,
      0x00, 0x03, 0xDB, 0x78,
      0x00, 0x03, 0xE4, 0x48,
      0x00, 0x03, 0xF3, 0x75,
      0x00, 0x03, 0xFB, 0x1D,
      0x00, 0x04, 0x13, 0x27,
      0x00, 0x04, 0x1C, 0xAC,
      0x00, 0x04, 0x28, 0xD5,
      0x00, 0x04, 0x2E, 0x69,
      0x00, 0x04, 0x3D, 0x88,
      0x00, 0x04, 0x4E, 0x71,
      0x00, 0x04, 0x54, 0xF2,
      0x00, 0x04, 0x67, 0x22,
      0x00, 0x04, 0x6F, 0xCD,
      0x00, 0x04, 0x7E, 0x75,
      0x00, 0x04, 0x80, 0x4B,
      0x00, 0x04, 0x90, 0x94,
      0x00, 0x04, 0x98, 0x59,
      0x00, 0x04, 0xA8, 0x78,
      0x00, 0x04, 0xB8, 0x79,
      0x00, 0x04, 0xC0, 0x64,
      0x00, 0x04, 0xCB, 0x7D,
      0x00, 0x04, 0xDA, 0x92,
      0x00, 0x04, 0xE9, 0x10,
      0x00, 0x04, 0xEF, 0xFB,
      0x00, 0x04, 0xF9, 0x20,
      0x00, 0x05, 0x08, 0x4B,
      0x00, 0x05, 0x12, 0x6F,
      0x00, 0x05, 0x1C, 0xEE,
      0x00, 0x05, 0x29, 0x96,
      0x00, 0x05, 0x41, 0x36,
      0x00, 0x05, 0x52, 0xBA,
      0x00, 0x05, 0x5B, 0x8D,
      0x00, 0x05, 0x6D, 0x3B,
      0x00, 0x05, 0x81, 0x43,
      0x00, 0x05, 0x93, 0x57,
      0x00, 0x05, 0x97, 0x93,
      0x00, 0x05, 0xA2, 0xD4,
      0x00, 0x05, 0xAE, 0x4B,
      0x00, 0x05, 0xBF, 0x31,
      0x00, 0x05, 0xC1, 0x28,
      0x00, 0x05, 0xCA, 0xD8,
      0x00, 0x05, 0xD5, 0x06,
      0x00, 0x05, 0xE1, 0x31,
      0x00, 0x05, 0xE4, 0x9C,
      0x00, 0x05, 0xEF, 0x8F,
      0x00, 0x05, 0xFC, 0xA5,
      0x00, 0x06, 0x10, 0x6A,
      0x00, 0x06, 0x16, 0x03,
      0x00, 0x06, 0x21, 0x58,
      0x00, 0x06, 0x25, 0xF5,
      0x00, 0x06, 0x36, 0x07,
      0x00, 0x06, 0x3B, 0xCA,
      0x00, 0x06, 0x49, 0x9D,
      0x00, 0x06, 0x4C, 0x88,
      0x00, 0x06, 0x63, 0xED,
      0x00, 0x06, 0x68, 0x65,
      0x00, 0x06, 0x7C, 0xEB,
      0x00, 0x06, 0x7F, 0xC6,
      0x00, 0x06, 0x83, 0xFE,
      0x00, 0x06, 0x90, 0x87,
      0x00, 0x06, 0x94, 0xC1,
      0x00, 0x06, 0x96, 0xEC,
      0x00, 0x06, 0x9A, 0xE1,
      0x00, 0x06, 0xA4, 0xB9,
      0x00, 0x06, 0xA7, 0x5D,
      0x00, 0x06, 0xAA, 0x7A,
      0x00, 0x06, 0xAE, 0x29,
      0x00, 0x06, 0xB7, 0xD8,
      0x00, 0x06, 0xBA, 0x6F,
      0x00, 0x06, 0xBE, 0x15,
      0x00, 0x06, 0xC1, 0x89,
      0x00, 0x06, 0xC7, 0xC1,
      0x00, 0x06, 0xCB, 0x1D,
      0x00, 0x06, 0xCE, 0x54,
      0x00, 0x06, 0xD1, 0x5D,
      0x00, 0x06, 0xD6, 0x2F,
      0x00, 0x06, 0xD9, 0xF0,
      0x00, 0x06, 0xDD, 0x21,
      0x00, 0x06, 0xE0, 0x8E,
      0x00, 0x06, 0xEE, 0xC4,
      0x00, 0x06, 0xF2, 0x72,
      0x00, 0x06, 0xF5, 0x7B,
      0x00, 0x06, 0xFA, 0x6C,
      0x00, 0x06, 0xFE, 0x0D,
      0x00, 0x07, 0x01, 0x90,
      0x00, 0x07, 0x04, 0x7F,
      0x00, 0x07, 0x11, 0xE4,
      0x00, 0x07, 0x15, 0xCC,
      0x00, 0x07, 0x1B, 0x09,
      0x00, 0x07, 0x1C, 0xCB,
      0x00, 0x07, 0x29, 0xFA,
      0x00, 0x07, 0x35, 0x07,
      0x00, 0x07, 0x42, 0xE1,
      0x00, 0x07, 0x50, 0x20,
      0x00, 0x07, 0x56, 0x5F,
      0x00, 0x07, 0x6A, 0xF8,
      0x00, 0x07, 0x73, 0x5D,
      0x00, 0x07, 0x7F, 0x9F,
      0x00, 0x07, 0x91, 0x15,
      0x00, 0x07, 0x99, 0x27,
      0x00, 0x07, 0xAA, 0xE0,
      0x00, 0x07, 0xB8, 0x8F,
      0x00, 0x07, 0xCC, 0x61,
      0x00, 0x07, 0xD3, 0x67,
      0x00, 0x07, 0xE9, 0x04,
      0x00, 0x07, 0xF0, 0xE7,
      0x00, 0x07, 0xFB, 0x61,
      0x00, 0x08, 0x06, 0x13,
      0x00, 0x08, 0x17, 0x39,
      0x00, 0x08, 0x25, 0xFF,
      0x00, 0x08, 0x2B, 0xBC,
      0x00, 0x08, 0x3A, 0x84,
      0x00, 0x08, 0x50, 0x23,
      0x00, 0x08, 0x57, 0x76,
      0x00, 0x08, 0x5A, 0x82,
      0x00, 0x08, 0x64, 0xFE,
      0x00, 0x08, 0x69, 0xFE,
      0x00, 0x08, 0x7A, 0x64,
      0x00, 0x08, 0x7F, 0xA5,
      0x00, 0x08, 0x8E, 0x56,
      0x00, 0x08, 0x93, 0x94,
      0x00, 0x08, 0xA2, 0x24,
      0x00, 0x08, 0xA7, 0x8A,
      0x00, 0x08, 0xAD, 0x27,
      0x00, 0x08, 0xB1, 0xCC,
      0x00, 0x08, 0xBC, 0x6D,
      0x00, 0x08, 0xC4, 0x1A,
      0x00, 0x08, 0xC9, 0xD9,
      0x00, 0x08, 0xCF, 0x2E,
      0x00, 0x08, 0xDD, 0x24,
      0x00, 0x08, 0xE5, 0x2E,
      0x00, 0x08, 0xF3, 0x32,
      0x00, 0x08, 0xFB, 0x00,
      0x00, 0x09, 0x12, 0x9F,
      0x00, 0x09, 0x23, 0x12,
      0x00, 0x09, 0x28, 0xA3,
      0x00, 0x09, 0x2F, 0xA9,
      0x00, 0x09, 0x3E, 0x48,
      0x00, 0x09, 0x4A, 0x21,
      0x00, 0x09, 0x4D, 0x57,
      0x00, 0x09, 0x56, 0x5E,
      0x00, 0x09, 0x60, 0x2D,
      0x00, 0x09, 0x6C, 0xEE,
      0x00, 0x09, 0x74, 0x5D,
      0x00, 0x09, 0x7D, 0xEB,
      0x00, 0x09, 0x86, 0x6E,
      0x00, 0x09, 0x92, 0x0C,
      0x00, 0x09, 0x9B, 0x60,
      0x00, 0x09, 0xA6, 0x25,
      0x00, 0x09, 0xB0, 0x1A,
      0x00, 0x09, 0xBA, 0x44,
      0x00, 0x09, 0xC2, 0x49,
      0x00, 0x09, 0xCB, 0xEF,
      0x00, 0x09, 0xD3, 0xDA,
    ])
    expect(stco(stcoBody)).toMatchSnapshot()
  })
})
