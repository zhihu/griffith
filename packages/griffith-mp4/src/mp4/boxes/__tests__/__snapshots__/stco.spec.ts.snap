// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stco box should get the stco box tree 1`] = `
Object {
  "flags": 0,
  "samples": Array [
    Object {
      "chunkOffset": 9271,
    },
    Object {
      "chunkOffset": 17100,
    },
    Object {
      "chunkOffset": 18907,
    },
    Object {
      "chunkOffset": 19687,
    },
    Object {
      "chunkOffset": 20620,
    },
    Object {
      "chunkOffset": 21497,
    },
    Object {
      "chunkOffset": 22053,
    },
    Object {
      "chunkOffset": 24886,
    },
    Object {
      "chunkOffset": 26431,
    },
    Object {
      "chunkOffset": 28820,
    },
    Object {
      "chunkOffset": 30035,
    },
    Object {
      "chunkOffset": 30844,
    },
    Object {
      "chunkOffset": 33143,
    },
    Object {
      "chunkOffset": 34350,
    },
    Object {
      "chunkOffset": 36920,
    },
    Object {
      "chunkOffset": 39062,
    },
    Object {
      "chunkOffset": 41286,
    },
    Object {
      "chunkOffset": 42037,
    },
    Object {
      "chunkOffset": 44603,
    },
    Object {
      "chunkOffset": 45924,
    },
    Object {
      "chunkOffset": 48376,
    },
    Object {
      "chunkOffset": 49143,
    },
    Object {
      "chunkOffset": 51148,
    },
    Object {
      "chunkOffset": 53374,
    },
    Object {
      "chunkOffset": 54076,
    },
    Object {
      "chunkOffset": 57669,
    },
    Object {
      "chunkOffset": 58843,
    },
    Object {
      "chunkOffset": 62664,
    },
    Object {
      "chunkOffset": 63767,
    },
    Object {
      "chunkOffset": 67470,
    },
    Object {
      "chunkOffset": 69128,
    },
    Object {
      "chunkOffset": 72383,
    },
    Object {
      "chunkOffset": 73555,
    },
    Object {
      "chunkOffset": 77178,
    },
    Object {
      "chunkOffset": 78767,
    },
    Object {
      "chunkOffset": 81464,
    },
    Object {
      "chunkOffset": 82310,
    },
    Object {
      "chunkOffset": 86310,
    },
    Object {
      "chunkOffset": 87740,
    },
    Object {
      "chunkOffset": 91399,
    },
    Object {
      "chunkOffset": 94972,
    },
    Object {
      "chunkOffset": 96682,
    },
    Object {
      "chunkOffset": 97407,
    },
    Object {
      "chunkOffset": 98718,
    },
    Object {
      "chunkOffset": 101880,
    },
    Object {
      "chunkOffset": 103175,
    },
    Object {
      "chunkOffset": 104039,
    },
    Object {
      "chunkOffset": 107934,
    },
    Object {
      "chunkOffset": 110766,
    },
    Object {
      "chunkOffset": 113238,
    },
    Object {
      "chunkOffset": 116079,
    },
    Object {
      "chunkOffset": 119097,
    },
    Object {
      "chunkOffset": 123741,
    },
    Object {
      "chunkOffset": 125035,
    },
    Object {
      "chunkOffset": 129874,
    },
    Object {
      "chunkOffset": 131570,
    },
    Object {
      "chunkOffset": 135846,
    },
    Object {
      "chunkOffset": 136993,
    },
    Object {
      "chunkOffset": 138148,
    },
    Object {
      "chunkOffset": 142437,
    },
    Object {
      "chunkOffset": 144185,
    },
    Object {
      "chunkOffset": 145524,
    },
    Object {
      "chunkOffset": 146251,
    },
    Object {
      "chunkOffset": 151152,
    },
    Object {
      "chunkOffset": 152436,
    },
    Object {
      "chunkOffset": 155417,
    },
    Object {
      "chunkOffset": 156614,
    },
    Object {
      "chunkOffset": 159192,
    },
    Object {
      "chunkOffset": 163646,
    },
    Object {
      "chunkOffset": 167265,
    },
    Object {
      "chunkOffset": 171828,
    },
    Object {
      "chunkOffset": 173210,
    },
    Object {
      "chunkOffset": 177546,
    },
    Object {
      "chunkOffset": 180920,
    },
    Object {
      "chunkOffset": 182628,
    },
    Object {
      "chunkOffset": 183641,
    },
    Object {
      "chunkOffset": 186507,
    },
    Object {
      "chunkOffset": 191728,
    },
    Object {
      "chunkOffset": 193198,
    },
    Object {
      "chunkOffset": 198175,
    },
    Object {
      "chunkOffset": 200415,
    },
    Object {
      "chunkOffset": 204704,
    },
    Object {
      "chunkOffset": 206103,
    },
    Object {
      "chunkOffset": 212481,
    },
    Object {
      "chunkOffset": 215138,
    },
    Object {
      "chunkOffset": 219409,
    },
    Object {
      "chunkOffset": 220296,
    },
    Object {
      "chunkOffset": 225139,
    },
    Object {
      "chunkOffset": 226942,
    },
    Object {
      "chunkOffset": 231789,
    },
    Object {
      "chunkOffset": 233748,
    },
    Object {
      "chunkOffset": 237361,
    },
    Object {
      "chunkOffset": 238898,
    },
    Object {
      "chunkOffset": 242423,
    },
    Object {
      "chunkOffset": 244295,
    },
    Object {
      "chunkOffset": 247541,
    },
    Object {
      "chunkOffset": 248856,
    },
    Object {
      "chunkOffset": 252792,
    },
    Object {
      "chunkOffset": 255048,
    },
    Object {
      "chunkOffset": 258933,
    },
    Object {
      "chunkOffset": 260893,
    },
    Object {
      "chunkOffset": 267047,
    },
    Object {
      "chunkOffset": 269484,
    },
    Object {
      "chunkOffset": 272597,
    },
    Object {
      "chunkOffset": 274025,
    },
    Object {
      "chunkOffset": 277896,
    },
    Object {
      "chunkOffset": 282225,
    },
    Object {
      "chunkOffset": 283890,
    },
    Object {
      "chunkOffset": 288546,
    },
    Object {
      "chunkOffset": 290765,
    },
    Object {
      "chunkOffset": 294517,
    },
    Object {
      "chunkOffset": 294987,
    },
    Object {
      "chunkOffset": 299156,
    },
    Object {
      "chunkOffset": 301145,
    },
    Object {
      "chunkOffset": 305272,
    },
    Object {
      "chunkOffset": 309369,
    },
    Object {
      "chunkOffset": 311396,
    },
    Object {
      "chunkOffset": 314237,
    },
    Object {
      "chunkOffset": 318098,
    },
    Object {
      "chunkOffset": 321808,
    },
    Object {
      "chunkOffset": 323579,
    },
    Object {
      "chunkOffset": 325920,
    },
    Object {
      "chunkOffset": 329803,
    },
    Object {
      "chunkOffset": 332399,
    },
    Object {
      "chunkOffset": 335086,
    },
    Object {
      "chunkOffset": 338326,
    },
    Object {
      "chunkOffset": 344374,
    },
    Object {
      "chunkOffset": 348858,
    },
    Object {
      "chunkOffset": 351117,
    },
    Object {
      "chunkOffset": 355643,
    },
    Object {
      "chunkOffset": 360771,
    },
    Object {
      "chunkOffset": 365399,
    },
    Object {
      "chunkOffset": 366483,
    },
    Object {
      "chunkOffset": 369364,
    },
    Object {
      "chunkOffset": 372299,
    },
    Object {
      "chunkOffset": 376625,
    },
    Object {
      "chunkOffset": 377128,
    },
    Object {
      "chunkOffset": 379608,
    },
    Object {
      "chunkOffset": 382214,
    },
    Object {
      "chunkOffset": 385329,
    },
    Object {
      "chunkOffset": 386204,
    },
    Object {
      "chunkOffset": 389007,
    },
    Object {
      "chunkOffset": 392357,
    },
    Object {
      "chunkOffset": 397418,
    },
    Object {
      "chunkOffset": 398851,
    },
    Object {
      "chunkOffset": 401752,
    },
    Object {
      "chunkOffset": 402933,
    },
    Object {
      "chunkOffset": 407047,
    },
    Object {
      "chunkOffset": 408522,
    },
    Object {
      "chunkOffset": 412061,
    },
    Object {
      "chunkOffset": 412808,
    },
    Object {
      "chunkOffset": 418797,
    },
    Object {
      "chunkOffset": 419941,
    },
    Object {
      "chunkOffset": 425195,
    },
    Object {
      "chunkOffset": 425926,
    },
    Object {
      "chunkOffset": 427006,
    },
    Object {
      "chunkOffset": 430215,
    },
    Object {
      "chunkOffset": 431297,
    },
    Object {
      "chunkOffset": 431852,
    },
    Object {
      "chunkOffset": 432865,
    },
    Object {
      "chunkOffset": 435385,
    },
    Object {
      "chunkOffset": 436061,
    },
    Object {
      "chunkOffset": 436858,
    },
    Object {
      "chunkOffset": 437801,
    },
    Object {
      "chunkOffset": 440280,
    },
    Object {
      "chunkOffset": 440943,
    },
    Object {
      "chunkOffset": 441877,
    },
    Object {
      "chunkOffset": 442761,
    },
    Object {
      "chunkOffset": 444353,
    },
    Object {
      "chunkOffset": 445213,
    },
    Object {
      "chunkOffset": 446036,
    },
    Object {
      "chunkOffset": 446813,
    },
    Object {
      "chunkOffset": 448047,
    },
    Object {
      "chunkOffset": 449008,
    },
    Object {
      "chunkOffset": 449825,
    },
    Object {
      "chunkOffset": 450702,
    },
    Object {
      "chunkOffset": 454340,
    },
    Object {
      "chunkOffset": 455282,
    },
    Object {
      "chunkOffset": 456059,
    },
    Object {
      "chunkOffset": 457324,
    },
    Object {
      "chunkOffset": 458253,
    },
    Object {
      "chunkOffset": 459152,
    },
    Object {
      "chunkOffset": 459903,
    },
    Object {
      "chunkOffset": 463332,
    },
    Object {
      "chunkOffset": 464332,
    },
    Object {
      "chunkOffset": 465673,
    },
    Object {
      "chunkOffset": 466123,
    },
    Object {
      "chunkOffset": 469498,
    },
    Object {
      "chunkOffset": 472327,
    },
    Object {
      "chunkOffset": 475873,
    },
    Object {
      "chunkOffset": 479264,
    },
    Object {
      "chunkOffset": 480863,
    },
    Object {
      "chunkOffset": 486136,
    },
    Object {
      "chunkOffset": 488285,
    },
    Object {
      "chunkOffset": 491423,
    },
    Object {
      "chunkOffset": 495893,
    },
    Object {
      "chunkOffset": 497959,
    },
    Object {
      "chunkOffset": 502496,
    },
    Object {
      "chunkOffset": 505999,
    },
    Object {
      "chunkOffset": 511073,
    },
    Object {
      "chunkOffset": 512871,
    },
    Object {
      "chunkOffset": 518404,
    },
    Object {
      "chunkOffset": 520423,
    },
    Object {
      "chunkOffset": 523105,
    },
    Object {
      "chunkOffset": 525843,
    },
    Object {
      "chunkOffset": 530233,
    },
    Object {
      "chunkOffset": 534015,
    },
    Object {
      "chunkOffset": 535484,
    },
    Object {
      "chunkOffset": 539268,
    },
    Object {
      "chunkOffset": 544803,
    },
    Object {
      "chunkOffset": 546678,
    },
    Object {
      "chunkOffset": 547458,
    },
    Object {
      "chunkOffset": 550142,
    },
    Object {
      "chunkOffset": 551422,
    },
    Object {
      "chunkOffset": 555620,
    },
    Object {
      "chunkOffset": 556965,
    },
    Object {
      "chunkOffset": 560726,
    },
    Object {
      "chunkOffset": 562068,
    },
    Object {
      "chunkOffset": 565796,
    },
    Object {
      "chunkOffset": 567178,
    },
    Object {
      "chunkOffset": 568615,
    },
    Object {
      "chunkOffset": 569804,
    },
    Object {
      "chunkOffset": 572525,
    },
    Object {
      "chunkOffset": 574490,
    },
    Object {
      "chunkOffset": 575961,
    },
    Object {
      "chunkOffset": 577326,
    },
    Object {
      "chunkOffset": 580900,
    },
    Object {
      "chunkOffset": 582958,
    },
    Object {
      "chunkOffset": 586546,
    },
    Object {
      "chunkOffset": 588544,
    },
    Object {
      "chunkOffset": 594591,
    },
    Object {
      "chunkOffset": 598802,
    },
    Object {
      "chunkOffset": 600227,
    },
    Object {
      "chunkOffset": 602025,
    },
    Object {
      "chunkOffset": 605768,
    },
    Object {
      "chunkOffset": 608801,
    },
    Object {
      "chunkOffset": 609623,
    },
    Object {
      "chunkOffset": 611934,
    },
    Object {
      "chunkOffset": 614445,
    },
    Object {
      "chunkOffset": 617710,
    },
    Object {
      "chunkOffset": 619613,
    },
    Object {
      "chunkOffset": 622059,
    },
    Object {
      "chunkOffset": 624238,
    },
    Object {
      "chunkOffset": 627212,
    },
    Object {
      "chunkOffset": 629600,
    },
    Object {
      "chunkOffset": 632357,
    },
    Object {
      "chunkOffset": 634906,
    },
    Object {
      "chunkOffset": 637508,
    },
    Object {
      "chunkOffset": 639561,
    },
    Object {
      "chunkOffset": 642031,
    },
    Object {
      "chunkOffset": 644058,
    },
  ],
  "version": 0,
}
`;
